spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -167,6 +167,7 @@
     end
 
     it 'loader allows loading a function more than once' do
+      allow(File).to receive(:read).and_call_original
```

Comment:
@hlindberg I had to add these mocks in order for the loader tests to pass on ruby 2.6. I'm not sure why the old code passes on ruby 2.5 and earlier. I'm guessing something about ruby 2.6 is causing us to call `File.read` with additional arguments, and that these mocks should not be added. Can you take a look?

Reply:
The intent in the test is to get mocks for the explicit reads and leave all other reads alone. Seems the "leave all other alone" is different for some reason.
We do not want to call original for the explicit mocks as they return value or error.

Reply:
Hmm, @hlindberg could you try to run `bundle exec rspec spec/unit/pops/loaders/loaders_spec.rb` with ruby 2.6, from puppet#master and see why the behavior is different?

Reply:
Ah nevermind, [URI#host changed behavior in Ruby 2.6](https://bugs.ruby-lang.org/issues/14035), which broke `Puppet::Util.uri_to_path`, which broke the loaders. I have a fix for `uri_to_path`.

Reply:
Ruby 2.6 also changed `Pathname#read` to call `File.read` instead of `IO.read` which broke some of the loader tests. I've updated this in https://github.com/puppetlabs/puppet/pull/7494/commits/f6be4e64d92d590e7f47051fd6e1be839878eed8